File: angular-16-client/src/app/services/appointment.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {CalendarEvent} from "angular-calendar";
import {Observable} from "rxjs";
import {URL} from "../app.constants";
import {AppointmentDTO} from "../models/appointment.model";

@Injectable({
  providedIn: 'root'
})
export class AppointmentService {

  constructor(private http: HttpClient) {}

  getAppointments(): Observable<CalendarEvent[]> {
    return this.http.get<CalendarEvent[]>(`${URL}/appointments`);
  }

  addAppointment(appointment: AppointmentDTO): Observable<any> {
    return this.http.post<any>(`${URL}/appointments`, appointment);
  }
}

File: spring-boot-server/src/main/java/com/jmiko/reservations/service/AppointmentService.java
package com.jmiko.reservations.service;

import com.jmiko.reservations.dto.AppointmentDTO;
import com.jmiko.reservations.model.Appointment;
import org.springframework.http.ResponseEntity;

public interface AppointmentService {

    ResponseEntity<Appointment> addAppointment(AppointmentDTO appointmentDTO);

}
